src/Components/Aspire.Azure.Messaging.EventHubs/AspireEventHubsExtensions.cs

```diff
@@ -0,0 +1,129 @@
+
+    public static void AddAzureEventProcessorClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<AzureMessagingEventHubsSettings>? configureSettings = null,
+        Action<IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions>>? configureClientBuilder = null)
+    {
+        new EventProcessorClientComponent(builder.Configuration)
+            .AddClient(builder, DefaultConfigSectionName + nameof(EventProcessorClient),
+                configureSettings, configureClientBuilder, connectionName, serviceKey: null);
+    }
+
+    public static void AddKeyedAzureEventHubProcessorClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<AzureMessagingEventHubsSettings>? configureSettings = null,
+        Action<IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions>>? configureClientBuilder = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        string configurationSectionName = EventProcessorClientComponent
+            .GetKeyedConfigurationSectionName(name, DefaultConfigSectionName +
+                                                    nameof(EventProcessorClient));
+
+        new EventProcessorClientComponent(builder.Configuration)
+            .AddClient(builder, configurationSectionName, configureSettings,
+                configureClientBuilder, connectionName: name, serviceKey: name);
+    }
+
+    public static void AddAzureEventHubProducerClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<AzureMessagingEventHubsSettings>? configureSettings = null,
+        Action<IAzureClientBuilder<EventHubProducerClient, EventHubProducerClientOptions>>? configureClientBuilder = null)
+    {
+        new EventHubProducerClientComponent()
+            .AddClient(builder, DefaultConfigSectionName + nameof(EventHubProducerClient),
+                configureSettings, configureClientBuilder, connectionName, serviceKey: null);
+    }
+
+    public static void AddKeyedAzureEventHubProducerClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<AzureMessagingEventHubsSettings>? configureSettings = null,
+        Action<IAzureClientBuilder<EventHubProducerClient, EventHubProducerClientOptions>>? configureClientBuilder = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        string configurationSectionName = EventHubProducerClientComponent
+            .GetKeyedConfigurationSectionName(name, DefaultConfigSectionName +
+                                                    nameof(EventHubProducerClient));
+
+        new EventHubProducerClientComponent()
+            .AddClient(builder, configurationSectionName, configureSettings,
+                configureClientBuilder, connectionName: name, serviceKey: name);
+    }
+
+    public static void AddAzureEventHubConsumerClient(
```

Comment:
We should include the PartitionReceiver as well.

Reply:
Working on this now.

Reply:
Done